src/applications/check-in/travel-claim/pages/travel-mileage/travelMileage.unit.spec.js

```diff
@@ -0,0 +1,283 @@
+          <TravelMileage />
+        </CheckInProvider>,
+      );
+      expect(component.getByTestId('[]')).to.exist;
+    });
+    it('auto selects facilities based on redux value', () => {
+      const store = {
+        eligibleToFile: [
+          {
+            stationNo: '500',
+            startTime: '2021-08-19T13:56:31',
+          },
+          {
+            stationNo: '600',
+            startTime: '2021-08-19T13:56:31',
+          },
+        ],
+        facilitiesToFile: [
+          {
+            stationNo: '500',
+            startTime: '2021-08-19T13:56:31',
+            multipleAppointments: false,
+          },
+        ],
+      };
+      const expectedStateValue = [
+        {
+          stationNo: '500',
+          startTime: '2021-08-19T13:56:31',
+          multipleAppointments: false,
+        },
+      ];
+      const component = render(
+        <CheckInProvider store={store}>
+          <TravelMileage />
+        </CheckInProvider>,
+      );
+      expect(component.getByTestId(JSON.stringify(expectedStateValue))).to
+        .exist;
+    });
+    describe('continueClick', () => {
+      it('continues to the next page with a selection', () => {
+        const store = {
+          formPages: ['travel-info', 'travel-mileage', 'travel-vehicle'],
+          eligibleToFile: [
+            {
+              stationNo: '500',
+              startTime: '2021-08-19T13:56:31',
+            },
+          ],
+        };
+        const push = sinon.spy();
+        const component = render(
+          <CheckInProvider
+            store={store}
+            router={{
+              push,
+              currentPage: '/travel-mileage',
+              params: {},
+            }}
+          >
+            <TravelMileage />
+          </CheckInProvider>,
+        );
+        const link = component.getByTestId('continue-button');
+        fireEvent.click(link);
+        expect(push.calledWith('travel-vehicle')).to.be.true;
+      });
+      it('sets error state if continue clicked with no selection', () => {
+        const store = {
+          formPages: ['travel-info', 'travel-mileage', 'travel-vehicle'],
+          eligibleToFile: [
+            {
+              stationNo: '500',
+              startTime: '2021-08-19T13:56:31',
+            },
+            {
+              stationNo: '600',
+              startTime: '2021-08-19T13:56:31',
+            },
+          ],
+        };
+        const push = sinon.spy();
+        const component = render(
+          <CheckInProvider
+            store={store}
+            router={{
+              push,
+              currentPage: '/travel-mileage',
+              params: {},
+            }}
+          >
+            <TravelMileage />
+          </CheckInProvider>,
+        );
+        const link = component.getByTestId('continue-button');
+        fireEvent.click(link);
+        expect(push.notCalled).to.be.true;
+      });
+    });
+    // This test is currently impossible due to shadowDOM
```

Comment:
Maybe we can remove the commented out code and create tech debt ticket and drop this in as an example